### PR TITLE
Run multicluster test in kind

### PR DESCRIPTION
**Pull request by howardjohn**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: prow/istio-pilot-multicluster-e2e.sh [hunk 1]
```diff
@@ -14,14 +14,108 @@
...
+# Create two clusters. Explicitly delcare the subnet so we can connect the two later
+kind create cluster --name a --config <(echo "${CONFIG_A}")
+kind create cluster --name b --config <(echo "${CONFIG_B}")
+
+export HUB=istio-testing
+export TAG=istio-testing
+# Build just the images needed for the tests
+for image in pilot proxyv2 proxy_init app test_policybackend mixer citadel galley sidecar_injector kubectl node-agent-k8s; do
+   make docker.${image}
+done
```
Comment (tiswanso) [inline prow/istio-pilot-multicluster-e2e.sh]: I'm surprised we have to do this in here.  We can't use images from a build stage, ie. via `docker pull ...`?

Comment (howardjohn) [reply prow/istio-pilot-multicluster-e2e.sh]: Prow doesn't support pipelines/ordering of tests, so each one has to build everything.. it's very wasteful

**Comment by howardjohn** [comment]
> Generally looks good to me but you might also want to pull the multicluster stuff out of cluster_lib.sh if you're going to do it all here. Or maybe add comments in this file regarding the move to kind v. the boskos / mason compatible content in cluster_lib.sh. Do we want to rename and leave an on-demand non-Kind test?

Once I get this passing there is definitely some cleanup that can be done. I don't think there are any plans to keep support for boskos around, it costs a ton of money to have all the clusters sitting around

**Comment by howardjohn** [comment]
> Generally looks good to me but you might also want to pull the multicluster stuff out of cluster_lib.sh if you're going to do it all here.

Yeah I definitely think this could use some cleanup, just trying to get things passing for now.

> I'm wondering if the cert setting might be needed and the clusters[0].cluster.server value in kubeconf change to the IP in the cert.

Seems plausible, I'll try that out when I get a chance. Thanks for the suggestion!

If anyone else has time to kick tires on this feel free to as well

**Comment by howardjohn** [comment]
Tests are failing because requests from `t` -> `a` are going through the inbound passthrough cluster on `a` and not getting any Envoy headers. It looks like other requests are also going through inbound cluster, but must get the headers set on outbonud

**Comment by howardjohn** [comment]
@clarketm tests aren't starting here for some reason

**Comment by clarketm** [comment]
@howardjohn - there was a hiccup with plank. Everything seems to be back online. 👍

**Comment by howardjohn** [comment]
Thanks! looks like some gcp issues so not too surprising

**Comment by howardjohn** [comment]
Im pretty sure I got this working but it is broken by https://github.com/istio/istio/issues/18211

I'll need to cleanup the script as well.

Thanks @tiswanso for the tip, that was the problem, could not reach remote API server

File: prow/istio-pilot-multicluster-e2e.sh [hunk 2]
```diff
@@ -14,14 +14,120 @@
...
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  podSubnet: ${SUBNET_B}
+  serviceSubnet: 10.255.10.0/24
+EOF
+)
+
+function cleanup_kind_cluster() {
+  echo "Test exited with exit code $?."
+  kind export logs --name "${1}" "${ARTIFACTS}/kind-${1}/"
```
Comment (incfly) [inline prow/istio-pilot-multicluster-e2e.sh]: this fails at locally if `ARTIFACTS`  is not set. I assume it's some prow bash library set it up. and left the cluster undeleted.

**Comment by incfly** [comment]
might be worth to name the cluster and primary/remote to match the test code.

**Comment by howardjohn** [comment]
> might be worth to name the cluster and primary/remote to match the test code.

Good idea

> ARTIFACTs

Good catch, I'll make it default to tmp dir. I need to clean this script up to reuse the existing kind functions but was just hacking some stuff together for now

**Comment by howardjohn** [comment]
@tiswanso ptal, I think this is ready

**Pull request merged**